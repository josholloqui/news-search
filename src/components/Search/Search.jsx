import React from 'react';
import PropTypes from 'prop-types';

const Search = ({ term, onChange, onSubmit }) => (
  <>
    <form onSubmit={onSubmit} >
      <label>
        <input
          id="term"
          type="text"
          name="term"
          value={term}
          onChange={onChange}
        />
      </label>
      <button>SEARCH</button>
    </form>
  </>
);

Search.propTypes = {
  term: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired
};

export default Search;
