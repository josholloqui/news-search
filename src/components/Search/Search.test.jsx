import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Search from './Search';

describe('Search component', () => {
  afterEach(() => cleanup());
  it('renders search', () => {
    const { asFragment } = render(
      <Search
        term="chicken"
        onChange={() => {}}
        onSubmit={() => {}}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
