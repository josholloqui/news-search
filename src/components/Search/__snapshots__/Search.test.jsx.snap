// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component renders search 1`] = `
<DocumentFragment>
  <form>
    <label>
      <input
        id="term"
        name="term"
        type="text"
        value="chicken"
      />
    </label>
    <button>
      SEARCH
    </button>
  </form>
</DocumentFragment>
`;
